### PR TITLE
Fix improper documentation on casting non_exhaustive enums

### PR DESCRIPTION
**Pull request by mixy1**
Added a more complete example that explains what is allowed and what's not allowed when the non_exhaustive attribute is in use.

### CODE REVIEW

File: src/attributes/type_system.md
Comment (obi1kenobi) [inline src/attributes/type_system.md:166]: I'm not a maintainer of the reference, so this will need a review from someone else as well.

I'm not sure if casting any other type other than an enum is ever allowed in Rust. I can't think of any, and if there are some, it might be good to mention them.

A quick suggestion to avoid starting with `However` in two nearby paragraphs:
```suggestion
It's also not allowed to use numeric casts (`as`) on enums that contain any non-exhaustive variants.
```

Comment (mixy1) [reply src/attributes/type_system.md:166]: whoops completely missed the duplication of "However" 😬. Yeah I don't think anything other than an enum can be cast as well. I'll commit the suggestion.

**Review by ehuss — APPROVED** [review]
Thanks! Not sure how I missed this in the original PR.

I pushed some tweaks to the examples just to clarify some things.

**Pull request merged**
